allensdk/brain_observatory/behavior/behavior_project_cache.py

```diff
@@ -0,0 +1,371 @@
+
+        # Possibly explode and reindex
+        if by == "ophys_session_id":
+            pass
+        elif by == "ophys_experiment_id":
+            sessions = (sessions.reset_index()
+                        .explode("ophys_experiment_id")
+                        .set_index("ophys_experiment_id"))
+        else:
+            self.logger.warning(
+                f"Invalid value for `by`, '{by}', passed to get_session_table."
+                " Valid choices for `by` are 'ophys_experiment_id' and "
+                "'ophys_session_id'.")
+        return sessions
+
+    def add_manifest_paths(self, manifest_builder):
+        manifest_builder = super().add_manifest_paths(manifest_builder)
+        for key, config in self.MANIFEST_CONFIG.items():
+            manifest_builder.add_path(key, **config)
+        return manifest_builder
+
+    def get_experiment_table(
+            self,
+            suppress: Optional[List[str]] = None) -> pd.DataFrame:
+        """
+        Return summary table of all ophys_experiment_ids in the database.
+        :param suppress: optional list of columns to drop from the resulting
+            dataframe.
+        :type suppress: list of str
+        :rtype: pd.DataFrame
+        """
+        write_csv = partial(
+            _write_csv,
+            array_fields=["reporter_line", "driver_line"])
+        read_csv = partial(
+            _read_csv, index_col="ophys_experiment_id",
+            array_fields=["reporter_line", "driver_line"],
+            array_types=[str, str])
+        experiments = self._get_session_summary(
+            self.fetch_api.get_experiment_table, self.OPHYS_EXPERIMENTS_KEY,
+            write_csv, read_csv)
+        experiments = experiments.rename(columns={"genotype": "full_genotype"})
+        if suppress:
+            experiments.drop(columns=suppress, inplace=True, errors="ignore")
+        return experiments
+
+    def get_behavior_session_table(
+            self,
+            suppress: Optional[List[str]] = None) -> pd.DataFrame:
+        """
+        Return summary table of all behavior_session_ids in the database.
+        :param suppress: optional list of columns to drop from the resulting
+            dataframe.
+        :type suppress: list of str
+        :rtype: pd.DataFrame
+        """
+        read_csv = partial(
+            _read_csv, index_col="behavior_session_id",
+            array_fields=["reporter_line", "driver_line"],
+            array_types=[str, str])
+        write_csv = partial(
+            _write_csv, array_fields=["reporter_line", "driver_line"])
+        sessions = self._get_session_summary(
+            self.fetch_api.get_behavior_only_session_table,
+            self.BEHAVIOR_SESSIONS_KEY, write_csv, read_csv)
+        sessions = sessions.rename(columns={"genotype": "full_genotype"})
+        if suppress:
+            sessions.drop(columns=suppress, inplace=True, errors="ignore")
+        return sessions
+
+    def _get_session_summary(
```

Comment:
this is a handy method but not really specific to session summary data.

Reply:
Prior to some code restructuring this helper method really decreased the verbosity of the other calls. I agree it's not necessary anymore